File: java/0003_Longest_Substring_Without_Repeating_Characters.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set set = new HashSet();
        int n = s.length();
        int begin = 0, end = 0;
        int ans = 0;

        while(end < n) {
            if(set.contains(s.charAt(end))) {
                set.remove(s.charAt(begin++));
            } else {
                set.add(s.charAt(end++));
                ans = Math.max(ans, end - begin);
            }
        }
        return ans;
    }
}

File: kotlin/0003_Longest_Substring_Without_Repeating_Characters.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        val set = hashSetOf<Char>()
        val n = s.length
        var begin = 0
        var end = 0
        var answer = 0

        while (end < n) {
            if (s[end] in set) {
                set.remove(s[begin++])
            } else {
                set.add(s[end++])
                answer = Integer.max(answer, end - begin)
            }
        }
        return answer
    }
}